### DISCUSSION TITLE
Store motion value (useTransform + useSpring) in zustand store not working

### DISCUSSION THREAD

**Discussion by trompx** [motiondivision/motion#1566] [Q&A]
Hello,
I am creating a smooth scroll with framer motion doing the following:
```js
  ...
  const { scrollY } = useViewportScroll();
  const transform = useTransform(scrollY, (value) => factor * value);
  const framerScrollY = useSpring(transform, framerSmoothScrollPhysics);
  setFramerScrollY(framerScrollY) // zustand method to update the store
  ...
```

I want to store the smoothScrollY in a zustand store so I can access this value accross multiple components.
I am using the `subscribe` method of zustand to create a hook and get store value update without triggering a rerender.
In a raf loop, I can confirm that the value of the motion in the store is updated, but when I try to use it with framer useTransform, it seems the value is not used:
```js
const framerScrollY = useFramerScrollY(); // my hook that gets the value from the zustand store that doesn't trigger a rerender
// PROGRESS REMAINS 0
const progress = useTransform(framerScrollY, bounds, [0, 1], { clamp });
// but framerScrollY.current.current is correctly updated
useRaf(() => {
  console.log(progress.current, framerScrollY.current.current);
});
```
However, if I apply the same logic as I did to create the smooth effect, the value gets updated:
```js
  const { scrollY } = useViewportScroll();
  const transform = useTransform(scrollY, (value) => factor * value);
  const framerScrollY = useSpring(transform, framerSmoothScrollPhysics);
  // PROGRESS IS UPDATED
  const progress = useTransform(framerScrollY, bounds, [0, 1], { clamp });
```

Is there a way to store the motion value in a react/zustand context/store and get framer to update its value or the way to do it is to factorize the part of the code that create the smooth effect and use it as a base for all other components depending on it?

Thanks in advance for the help.

**Comment by harshagarwalnyu** [comment]
Motion values are **reactive subscriptions**, not plain JS values. Storing them in Zustand (which expects serializable state) is the wrong pattern and won't work reliably — Zustand re-renders subscribers synchronously, but motion values update outside React's render cycle.

**Correct approach: share the motion value via React Context or module scope — NOT Zustand**

```tsx
// scrollStore.ts — module-level singleton (outside React)
import { useViewportScroll, useTransform, useSpring } from 'framer-motion';
import { createContext, useContext } from 'react';

const PHYSICS = { mass: 0.27, stiffness: 100, damping: 30 };

// Create context to share across components
export const ScrollContext = createContext(null);

export function ScrollProvider({ children, factor = 1 }) {
  const { scrollY } = useViewportScroll();
  const transform = useTransform(scrollY, (v) => factor * v);
  const framerScrollY = useSpring(transform, PHYSICS);

  return (
    <ScrollContext.Provider value={framerScrollY}>
      {children}
    </ScrollContext.Provider>
  );
}

export const useFramerScrollY = () => useContext(ScrollContext);
```

Then in any component:
```tsx
const framerScrollY = useFramerScrollY();

useEffect(() => {
  return framerScrollY.on('change', (v) => {
    // use v — does NOT trigger re-render
  });
}, [framerScrollY]);

// Or drive a motion element:
<motion.div style={{ y: framerScrollY }} />
```

**If you specifically need a Zustand value (e.g., for non-motion consumers):**
```tsx
useEffect(() => {
  return framerScrollY.on('change', (v) => {
    useScrollStore.setState({ scrollY: v }); // push scalar to Zustand
  });
}, [framerScrollY]);
```

This keeps motion values in React/Framer land and only syncs scalar numbers to Zustand — no serialization issues, no stale subscription bugs.
